### PR TITLE
Accept tf.layers compatible argument names

### PR DESCRIPTION
**Pull request by ppwwyyxx**
#627 

Both old and tf.layers argument naming should be supported now. Here are some python hacks that seems to work.

BatchNorm TBD.

### CODE REVIEW

**Comment by Skylion007** [comment]
Ouch, on one hand great job for maintaining backwards compatibility. On the other hand, it's big blow to readability and documentation. ;-; Maintaining two separate keyword lists seems pretty awful and some of these hacks definitely don't follow [the principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Maybe a function decorator that would add an deprecation warning would be more readable? Perhaps something like the decorator in the top voted answer to [this](https://stackoverflow.com/questions/35567753/python-click-library-rename-argument) StackOverflow question?

**Comment by ppwwyyxx** [comment]
> readability and documentation.

I expect people to read tf.layers documentation instead of mine. I could copy docs from there. I could also add a new argument list plus **kwargs.

>  the principle of least astonishment.

How does this astonish a user who doesn't go and read source code?

The click library seems to only work for command line args if I'm not mistaken. Also what I need to do is not only rename.

**Comment by Skylion007** [comment]
Sorry, that Stackoverflow comment has nothing to do with the click library, but rather adding functionality that doesn't exist within it. I'll post the relevant portions.

```python
import functools

def rename_kwargs(**replacements):
    def actual_decorator(func):
        @functools.wraps(func)
        def decorated_func(*args, **kwargs):
            for internal_arg, external_arg in replacements.iteritems():
                if external_arg in kwargs:
                    kwargs[internal_arg] = kwargs.pop(external_arg)
                return func(*args, **kwargs)
        return decorated_func
    return actual_decorator


if __name__ == '__main__':

    @rename_kwargs(different_arg='format')
    def tester(different_arg):
        print different_arg

    tester(format='test value')
```
In the above snippet @rename_kwargs renames the keyword argument "format" to "different_arg" automatically. The decorator supports multiple key values passed in via a dict. You can augment the decorator to further modify the arguments as needed. That way when you decide to fully deprecate the old arguments you can just remove the decorators and the code will be readable and updated. Additionally, the decorator can be reused for all tf.layers argument translations and remove unnecessary code duplication. Isn't this how Tensorflow handles API changes anyhow?

> How does this astonish a user who doesn't go and read source code?
This is more about astonishing contributors like myself and others. It just adds pretty nasty complexity and adds to the technical debt. I definitely agree we should change the arguments, but there are better ways to do that.

Furthermore, supporting two separate versions of command arguments could get confusing. 

At the very least, do we want our examples to break whenever Tensorflow changes a default option if we rely on them for the defaults? One of the best parts of Tensorpack are the clean and well maintained examples that have excellent default values for performance.

**Comment by ppwwyyxx** [comment]
Decorator looks good. I'll give it a try.

Tensorflow promises to not do breaking change so using it is fine. It's their job to not break things and I'd rather not care about it.

**Comment by Skylion007** [comment]
The new changes with the decorator look good. Minor style point is that parse_args might not be the most descriptive name for what the decorator does. Otherwise, looks good.
